adam/learner/__init__.py

```diff
@@ -167,3 +172,27 @@ def graph_without_learner(perception_graph: PerceptionGraph) -> PerceptionGraph:
         islands = list(isolates(graph))
         graph.remove_nodes_from(islands)
     return PerceptionGraph(graph)
+
+
+class NewStyleLearner(ABC):
```

Comment:
@spigo900 : Please add docstrings for these methods.

Reply:
@gabbard Done. Should I also add a docstring for the class?

Reply:
Not yet - it's probably going to get collapsed with another one before this merges.